eisbuk-admin/src/components/atoms/AttendanceCard/__tests__/AttendanceCard.test.tsx

```diff
@@ -1,17 +1,28 @@
 import React from "react";
 import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
 
 import AttendanceCard from "../AttendanceCard";
 
-// import * as attendanceOperations from "@/store/actions/attendanceOperations";
+import * as attendanceOperations from "@/store/actions/attendanceOperations";
 import { Category } from "eisbuk-shared";
 import { dummySlot } from "../__testData__/dummyData";
 
 /**
  * Mocked `markAttendance` function, expect it to have been called when needed
  */
-// const mockMarkAttendance = jest.spyOn(attendanceOperations, "markAttendance");
+const mockMarkAttendance = jest.spyOn(attendanceOperations, "markAttendance");
 
+mockMarkAttendance.mockImplementation(
+  (customerId: string, attended: boolean) => ({
+    customerId,
+    attended,
+  })
+);
```

Comment:
Ok, so no need to store mockMarkAttendance in a variable any more (as we're not using it for assertions and mock implementation is constant throughout the tests). you could just write this:
```typescript
jest.spyOn(attendanceOperations, "markAttendance").mockImplementation(<mockImplementation>)
```
However, it would be a good idea to store mock implementation. as such:

```typescript
const mockMarkAttImplementation = (customerId: string, attended: boolean) => ({ customerId, attended })
```
This is good because we can use it further and write tests in such a way that:
1. We're eliminating chances for typos
2. The tests are conveying what we're testing with more accuracy

The way we would write the test would be:
```typescript
// first declare the mockImplementation
const mockMarkAttImplementation = (customerId: string, attended: boolean) => ({ customerId, attended })

// mock the implementation of the function within the module to use previously saved implementation
jest.spyOn(attendanceOperations, "markAttendance").mockImplementation(mockMarkAttImplementation)

// use mocked implementation to write more explicit test
expect(mockDispatch).toHaveBeenCalledWith(mockMarkAttImplementation("heisebnerg", true))
```